presto-spark-base/src/main/java/com/facebook/presto/spark/execution/nativeprocess/NativeExecutionProcess.java

```diff
@@ -263,12 +263,18 @@ private static URI getBaseUriWithPort(URI baseUri, int port)
     }
 
     private static int getAvailableTcpPort()
-            throws IOException
     {
-        ServerSocket socket = new ServerSocket(0);
-        int port = socket.getLocalPort();
-        socket.close();
-        return port;
+        try {
+            ServerSocket socket = new ServerSocket(0);
+            int port = socket.getLocalPort();
+            socket.close();
+            return port;
+        }
+        catch (Exception ex) {
+            // Something is wrong with the executor
+            throw new PrestoSparkFatalException("Failed to acquire port on host",
```

Comment:
nit: Do you want to include the original exception as the `PrestoSparkFatalException` cause? It may potentially include some useful insights on why the port allocation failed.

Reply:
Makes sense. Updated